metpy/calc/indices.py

```diff
@@ -305,3 +305,63 @@ def significant_tornado(sbcape, surface_based_lcl_height, storm_helicity_1km, sh
     return ((sbcape / (1500. * units('J/kg'))) *
             surface_based_lcl_height *
             (storm_helicity_1km / (150. * units('m^2/s^2'))) * shear_6km)
+
+
+@exporter.export
+@check_units('[pressure]', '[speed]', '[speed]', '[length]', '[speed]', '[speed]')
+def critical_angle(pressure, u, v, heights, stormu, stormv):
+    r"""Calculate the critical angle.
+
+    The critical angle is the angle between the 10m storm-relative inflow vector
+    and the 10m-500m shear vector. A critical angle near 90 degrees indicates
+    that a storm in this environment on the indicated storm motion vector
+    is likely ingesting purely streamwise vorticity into its updraft, and Esterheld and
+    Giuliano (2008) showed that significantly tornadic supercells tend to occur in environments
+    with critical angles near 90 degrees.
+
+    Parameters
+    ----------
+    pressure : `pint.Quantity`
+        Pressures from sounding.
```

Comment:
Would this function work on a grid?

Reply:
I think I would probably have to loop over the grid with it, since it uses get_layer.

Reply:
Fair enough.